src/engraving/rendering/score/horizontalspacing.cpp

```diff
@@ -469,6 +483,39 @@ void HorizontalSpacing::checkLyricsAgainstRightMargin(std::vector<SegmentPositio
     }
 }
 
+void HorizontalSpacing::checkLargeTimeSigAgainstRightMargin(std::vector<SegmentPosition>& segPositions)
+{
+    SegmentPosition& cautionaryTSSegPos = segPositions.back();
+    if (!cautionaryTSSegPos.segment->hasTimeSigAboveStaves()) {
+        return;
+    }
+
+    const MStyle& style = cautionaryTSSegPos.segment->style();
+    TimeSigVSMargin timeSigVSMargin = style.styleB(Sid::timeSigCenterOnBarline)
+                                      ? style.styleV(Sid::timeSigVSMarginCentered).value<TimeSigVSMargin>()
+                                      : style.styleV(Sid::timeSigVSMarginNonCentered).value<TimeSigVSMargin>();
+    if (timeSigVSMargin != TimeSigVSMargin::RIGHT_ALIGN_TO_BARLINE) {
+        return;
+    }
+
+    double xMax = 0.0;
+    for (auto iter = segPositions.rbegin(); iter != segPositions.rend(); ++iter) {
+        if (iter->segment->isType(SegmentType::EndBarLine | SegmentType::KeySigAnnounce)) {
+            xMax = std::max(xMax, iter->xPosInSystemCoords + iter->segment->minRight());
+        }
+    }
+
+    if (xMax == 0.0) {

```

Comment:
Do we need a fuzzy compare here, or is the result always going to be either the literal `0.0` you assign to it above or something that's nowhere near zero?